### PR TITLE
feat(operator): add 'fart' operator

### PR DESCRIPTION
**Pull request by jsayol**
**Description:**
First of all, I'm sorry. This is all @ladyleet's fault: https://twitter.com/ladyleet/status/816416159247593473

Adds the "fart" operator.

The tests *should* be passing but either jasmine or some function in between doesn't like the Unicode emoji, for some weird reason. And yes, "weird" is certainly a debatable term in the context of this PR.

Checklist:
- [x] Add the operator to Rx
- [x] It must have a `-spec.ts` tests file covering the canonical corner cases, with marble diagram tests
- [x] If possible, write a `asDiagram` test case too, for PNG diagram generation purposes
- [x] The spec file should have a type definition test at the end of the spec to verify type definition for various use cases
- [x] The operator must be documented in JSDoc style in the implementation file, including also the PNG marble diagram image
- [x] The operator should be listed in `doc/operators.md` in a category of operators
- [x] It should also be inserted in the operator decision tree file `doc/decision-tree-widget/tree.yml`

P.S.: I don't actually expect you to merge this PR but I figured if I was going to do it, I'd better do it right.

### CODE REVIEW

File: spec/operators/fart-spec.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    const expected = '|';
+
+    expectObservable(a.fart()).toBe(expected);
+    expectSubscriptions(a.subscriptions).toBe(asubs);
+  });
+
+  it('should map twice', () => {
+    const a = hot('-00----11-^-22---33--44-55--66--77-88-|');
+    const asubs =         '^                    !';
+    const expected =      '--💩💨---💩💨--💩💨-💩💨--💩💨--💩💨-💩💨-|';
```
Comment (blesh) [inline spec/operators/fart-spec.ts]: OMG hahahahaha

**Comment by blesh** [comment]
I TOTALLY love the spirit of this PR... but closing to prevent copycats (that might cause us more work). haha...

💩💨

**Comment by jsayol** [comment]
Haha, I would have been worried if this actually made it into master 😂

:hankey::dash:
